src/entity/find-entities.ts

```diff
@@ -86,5 +86,37 @@ export async function findEntities(
   }
 
   // Flatten the nested list
-  return Array.prototype.concat.apply([], nestedEntities);
+  let unorderedEntities = Array.prototype.concat.apply([], nestedEntities);
+  let orderedEntities = orderEntities(document, unorderedEntities);
+
+  return orderedEntities;
+}
+
+function orderEntities(
+  document: Document<any, any>,
+  unorderedEntities: Descriptor[]): Descriptor[] {
+
+  let entitiesByNode = new Map<any, Descriptor[]>();
+  let orderedEntities: Descriptor[] = [];
```

Comment:
This type should be `Descriptor[][]` shouldn't it? Is there something wrong with the type checker, or something wrong with my brain?